### PR TITLE
feat: added ability to search / filter out components

### PR DESCRIPTION
**Pull request by cadamsdev**
Media

msedge_V4USxjE96J

### CODE REVIEW

**Review by kof — APPROVED** [review]
amazing!

Approving to allow deployment, still in review process, will add more items

- [x] now we need to also add fuzzy search, check out how we use "matchSorter"
- [ ] need to fix SearchField ESC handling, seems broken, in fonts manager it should clear the seach first, not close the popover right away
- [x] add a separator line after search input
- [x] add autoFocus, so that when opening, input has caret right away

File: apps/builder/app/builder/features/sidebar-left/panels/components/components.tsx [hunk 1]
```diff
@@ -74,16 +89,56 @@ export const TabContent = ({ publish, onSetActiveTab }: TabContentProps) => {
...
                     .filter((meta: WsComponentMeta) => {
+                      const component = componentNamesByMeta.get(meta);
+                      if (
+                        searchText !== "" &&
+                        !component
+                          ?.toLowerCase()
+                          .includes(searchText.toLowerCase())
+                      ) {
+                        return false;
+                      }
```
Comment (cadamsdev) [inline apps/builder/app/builder/features/sidebar-left/panels/components/components.tsx:140]: TODO delete this part not needed

**Comment by cadamsdev** [comment]
> amazing!
> 
> Approving to allow deployment, still in review process, will add more items
> 
> * [ ]  now we need to also add fuzzy search, check out how we use "matchSorter"
> * [ ]  need to fix SearchField ESC handling, seems broken, in fonts manager it should clear the seach first, not close the popover right away
> * [ ]  add a separator line after search input
> * [ ]  add autoFocus, so that when opening, input has caret right away

- Added fuzzy search
- Added separator line after search input
- Added autoFocus

**Comment by kof** [comment]
Merged, follow up PR will:
- fix search field
- allow using keyboard arrows to navigate
- remove categories when searching

**Pull request merged**